avocado/core/suite.py

```diff
@@ -108,9 +108,9 @@ def _from_config_with_loader(cls, config, name=None):
         except (LoaderUnhandledReferenceError, LoaderError) as details:
             raise TestSuiteError(details)
 
-        return cls(name=name or str(uuid4()),
-                   config=config,
-                   tests=tests)
+        if name is None:
+            name = str(uuid4())
+        return cls(name=name, config=config, tests=tests)
```

Comment:
I would even go with something like `return cls(name = str(uuid4()) if not name else name, config=...` but I guess this is more standard.

Reply:
Because of the 80 columns limit, I think that is best to go with the current approach here.